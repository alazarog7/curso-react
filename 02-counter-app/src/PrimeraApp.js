import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

const PrimeraApp = ({nombre}) => {

    return (
        <>
            <h1> Hola {nombre}</h1>
            <p>Esta es mi presentaci√≥n</p>
        </>
    );
}

PrimeraApp.propTypes = {
    nombre : PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    nombre: 'Alejandro'
}

export default PrimeraApp;