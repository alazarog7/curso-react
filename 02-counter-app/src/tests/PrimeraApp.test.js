import React from 'react';
import {getByText, render} from '@testing-library/react';
import PrimeraApp from '../Primeraapp';
import { shallow } from 'enzyme';
describe('Pruebas en PrimeraApp', () => {
    //test('debe mostrar el mensa Hola Alejandro ', () => {
    //    const saludo = "Hola Alejandro";
    //    const wrapper = render(<PrimeraApp nombre="Alejandro"/>);
    //    // const {getByText} = render(<PrimeraApp nombre="Alejandro"/>);
    //    //console.log(wrapper);
    //    expect(wrapper.getByText(saludo)).toBeInTheDocument();
    //})

    test('debe mostrar el mensa Hola Alejandro', () => {
        const wrapper = shallow(<PrimeraApp nombre="alejandro"/>);
    
        expect(wrapper).toMatchSnapshot();
    })

    test('el subtitulo debe estar contenido dentro de un <p></p>',()=>{
        
        const wrapper = shallow(<PrimeraApp nombre="alejandro"/>);
        const subtituloTest = "Esta es mi presentaci√≥n";

        const subtitulo = wrapper.find('p').text();

        expect(subtitulo).toBe(subtituloTest);
        
    });
    


    
})
