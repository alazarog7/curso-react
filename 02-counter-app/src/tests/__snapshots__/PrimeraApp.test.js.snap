// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar el mensa Hola Alejandro 1`] = `
<Fragment>
  <h1>
     Hola 
    alejandro
  </h1>
  <p>
    Esta es mi presentaci√≥n
  </p>
</Fragment>
`;
